src/pydvl/influence/influence_calculator.py

```diff
@@ -0,0 +1,554 @@
+from abc import ABC, abstractmethod
+from math import prod
+from typing import (
+    Callable,
+    Generator,
+    Generic,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    TypeVar,
+)
+
+import distributed
+from dask import array as da
+from dask import delayed
+from numpy.typing import NDArray
+
+from .base_influence_model import (
+    InfluenceFunctionModel,
+    InfluenceType,
+    TensorType,
+    UnSupportedInfluenceTypeException,
+)
+
+__all__ = ["DaskInfluenceCalculator"]
+
+
+class DimensionChunksException(ValueError):
+    def __init__(self, chunks: Tuple[Tuple[int, ...], ...]):
+        msg = (
+            f"Array must be un-chunked in every dimension but the first, got {chunks=}"
+        )
+        super().__init__(msg)
+
+
+class UnalignedChunksException(ValueError):
+    def __init__(self, chunk_sizes_x: Tuple[int, ...], chunk_sizes_y: Tuple[int, ...]):
+        msg = (
+            f"Arrays x and y must have the same chunking in the first dimension, got {chunk_sizes_x=} "
+            f"and {chunk_sizes_y=}"
+        )
+        super().__init__(msg)
+
+
+class NumpyConverter(Generic[TensorType], ABC):
+    """
+    Base class for converting TensorType objects into numpy arrays and vice versa.
+    """
+
+    @abstractmethod
+    def to_numpy(self, x: TensorType) -> NDArray:
+        """Overwrite this method for converting a TensorType object into a numpy array"""
+
+    @abstractmethod
+    def from_numpy(self, x: NDArray) -> TensorType:
+        """Overwrite this method for converting a numpy array into a TensorType object"""
+
+
+class DaskInfluenceCalculator:
+    """
+    Compute influences over dask.Array collections. Depends on a batch computation model
+    of type [InfluenceFunctionModel][pydvl.influence.base_influence_model.InfluenceFunctionModel].
+    In addition, provide transformations from and to numpy,
+    corresponding to the tensor types of the batch computation model.
+    Args:
+        influence_function_model: instance of type
+            [InfluenceFunctionModel][pydvl.influence.base_influence_model.InfluenceFunctionModel], defines the
+            batch-wise computation model
+        numpy_converter: instance of type [NumpyConverter][pydvl.influence.influence_calculator.NumpyConverter], used to
+            convert between numpy arrays and TensorType objects needed to use the underlying model
+    """
+
+    def __init__(
+        self,
+        influence_function_model: InfluenceFunctionModel,
```

Comment:
Actually, I disagree that less is better. To quote a colleague of ours: Clarity over brevity. What would be the advantage to name it if_model?

Reply:
I also think longer names are better as they are more descriptive. When I read just "model" I think it is only the NN

Reply:
The problem is consistency. We can't prefix some variable names with types and not others. Since I agree with the confusion with the NN model, I propose to use `if_model` as a compromise, and to use the prefix `if_` consistently. 
Longer variable and class names are useful, I agree. But as I explain in another comment, they can be a mental drain and just get in the way. And it's not an either/or situation here (I'm not suggesting to call it `m`)

Reply:
As discussed, we leave it as it is.